include/RAJA/pattern/kernel/Tile.hpp

```diff
@@ -200,16 +205,43 @@ template <camp::idx_t ArgumentId,
           typename... EnclosedStmts>
 struct StatementExecutor<statement::
                              Tile<ArgumentId, TPol, EPol, EnclosedStmts...>> {
+  template <typename Data>
+  static RAJA_INLINE void exec(Data &data)
+  {
+    // Get the segment we are going to tile
+    auto const &segment = camp::get<ArgumentId>(data.segment_tuple);
 
+    // Get the tiling policies chunk size
+    auto chunk_size = TPol::chunk_size;
 
+    // Create a tile iterator
+    IterableTiler<decltype(segment)> tiled_iterable(segment, chunk_size);
+
+    // Wrap in case forall_impl needs to thread_privatize
+    TileWrapper<ArgumentId, Data, EnclosedStmts...> tile_wrapper(data);
+
+    // Loop over tiles, executing enclosed statement list
+    forall_impl(EPol{}, tiled_iterable, tile_wrapper);
+
+    // Set range back to original values
+    camp::get<ArgumentId>(data.segment_tuple) = tiled_iterable.it;
+  }
+};
+
+template<camp::idx_t ArgumentId,
+  typename EPol,
+  typename... EnclosedStmts>
+struct StatementExecutor<statement::
+                             Tile<ArgumentId, statement::tile_dynamic<ArgumentId>, EPol, EnclosedStmts...>> {
   template <typename Data>
   static RAJA_INLINE void exec(Data &data)
   {
     // Get the segment we are going to tile
     auto const &segment = camp::get<ArgumentId>(data.segment_tuple);
 
     // Get the tiling policies chunk size
-    auto chunk_size = TPol::chunk_size;
+    auto chunk_size = camp::get<ArgumentId>(data.param_tuple);
+    std::cout << "Tile size is " << chunk_size << std::endl;
```

Comment:
:(

Reply:
Not ready to merge - this is just a WIP.